dotnet/samples/Concepts/ChatCompletion/ChatHistoryReducers/ChatCompletionServiceExtensions.cs

```diff
@@ -0,0 +1,369 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Runtime.CompilerServices;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.ChatCompletion;
+
+namespace ChatCompletion;
+
+/// <summary>
+/// Extensions methods for <see cref="IChatCompletionService"/>
+/// </summary>
+internal static class ChatCompletionServiceExtensions
+{
+    /// <summary>
+    /// Adds an wrapper to an instance of <see cref="IChatCompletionService"/> which will use
+    /// the provided instance of <see cref="IChatHistoryReducer"/> to reduce the size of
+    /// the <see cref="ChatHistory"/> before sending it to the model.
+    /// </summary>
+    /// <param name="service">Instance of <see cref="IChatCompletionService"/></param>
+    /// <param name="reducer">Instance of <see cref="IChatHistoryReducer"/></param>
+    public static IChatCompletionService UsingChatHistoryReducer(this IChatCompletionService service, IChatHistoryReducer reducer)
+    {
+        return new ChatCompletionServiceWithReducer(service, reducer);
+    }
+}
+
+/// <summary>
+/// Implementation of <see cref="IChatHistoryReducer"/> which truncates chat history to the provide truncated size.
+/// </summary>
+/// <remarks>
+/// The truncation process is triggered when the list length is great than the truncated size.
+/// </remarks>
+public sealed class TruncatingChatHistoryReducer : IChatHistoryReducer
+{
+    private readonly int _truncatedSize;
+
+    /// <summary>
+    /// Creates a new instance of <see cref="TruncatingChatHistoryReducer"/>.
+    /// </summary>
+    /// <param name="truncatedSize">The size of the chat history after truncation.</param>
+    public TruncatingChatHistoryReducer(int truncatedSize)
```

Comment:
I still think an optional _threshold_ is a useful option.  Do you really want tuncation to always occur on every invocation after `truncatedSize` is exceeded without providing any other less active option for the dev?>

Reply:
I think it's OK for this sample, the process of truncating to the last N messages is easier to understand, this is only a sample.